pyanaconda/bootloader.py

```diff
@@ -680,12 +683,16 @@ def set_stage1_device(self, devices):
                     self.stage1_device = device.parents[0]
                 else:
                     self.stage1_device = device
-
                 break
+            errors.extend(self.errors)
 
         if not self.stage1_device:
             self.reset()
-            raise BootLoaderError("failed to find a suitable stage1 device")
+            msg = "Failed to find a suitable stage1 device"
+            if errors:
+                msg = msg + ": " + "; ".join(errors)
+                self.errors = errors
```

Comment:
The self.errors (self is a Bootloader instance) will be gathered when creating error details for custom partitioning.
Here the errors are collected locally in errors variable because is_valid_stage1_device clears the self.errors when called [1] so older errors from the loop above would be lost.

[1] it may seem as not correct but there are already parts in the code (https://github.com/rhinstaller/anaconda/blob/rhel7-branch/pyanaconda/bootloader.py#L1333) that take it into account so I decided to deal with it locally to make the patch safer/isolated. For master it might be worthy to dig deeper.